### PR TITLE
SIGN-7489 - added TBS token, and API Url task params

### PR DESCRIPTION
**Pull request by volbobvol**
Re-introduced the ApiUrl parameter and TBS Token Credential ID in the pipeline steps
Made all three global fields (ApiUrl, TBS Credential ID, Org Id) required
If the step-provided values do not match the global ones, fail with an error message

### Testing done
Tested the following cases:
- Fields are not set neither at the global level nor at the pipeline level.  
- Fields are set only at the global level.  
- Fields are set only at the pipeline level.  
- Fields are set at both levels, and the pipeline-level values are identical to the global-level values.  
- Fields are set at both levels, but the pipeline-level values differ from the global-level values.  

### Submitter checklist
- [x] Make sure you are opening from a **topic/feature/bugfix branch** (right side) and not your main branch!
- [x] Ensure that the pull request title represents the desired changelog entry
- [x] Please describe what you did
- [x] Link to relevant issues in GitHub or Jira
- [x] Link to relevant pull requests, esp. upstream and downstream changes
- [x] Ensure you have provided tests - that demonstrates feature works or fixes the issue

### CODE REVIEW

File: src/main/java/io/jenkins/plugins/signpath/SignPathStepBase.java [hunk 1]
```diff
@@ -122,15 +150,45 @@ protected String ensureNotNull(String input, String name) throws SignPathStepInv
...
+    protected String getWithGlobalConfig(String stepLevelValue, Function<SignPathPluginGlobalConfiguration, String> globalValueGetter, String paramName) throws SignPathStepInvalidArgumentException {
+
+        SignPathPluginGlobalConfiguration config = GlobalConfiguration.all().get(SignPathPluginGlobalConfiguration.class);
+        if (config == null) {
+            throw new IllegalStateException("SignPathPluginGlobalConfiguration is not available. Ensure it is properly configured.");
+        }
+        String globalVal = globalValueGetter.apply(config);
+
+        if (globalVal == null || globalVal.isEmpty()) {
+            // global value is not set yet, fallback to the obsolete step-level value
```
Comment (Taha-cmd) [inline src/main/java/io/jenkins/plugins/signpath/SignPathStepBase.java:162]: This function doesn't know whether the step-value is obsolete or not (you probably mean deprecated here right?). Maybe rephrase the comment?

File: src/main/java/io/jenkins/plugins/signpath/SignPathStepBase.java [hunk 2]
```diff
@@ -122,15 +150,45 @@ protected String ensureNotNull(String input, String name) throws SignPathStepInv
...
+        if (stepLevelValue == null || stepLevelValue.isEmpty()) {
+            // step level value is not set, use global value
+            return globalVal;
+        }
+
+        // here we have both values set. We enforce a rule that step level value should be identical to global value
+        if (!stepLevelValue.equals(globalVal)) {
+            throw new SignPathStepInvalidArgumentException(
+                String.format(
+                    "Parameter '%s' is configured globally to '%s' and cannot be changed in pipeline to '%s'.",
```
Comment (Taha-cmd) [inline src/main/java/io/jenkins/plugins/signpath/SignPathStepBase.java:175]: `Parameter <param-name> is configured globally to <global-value> and cannot be changed in pipeline to <step-value>` kind of implies that it's about the actual values rather than overriding a global parameter with a step value. Maybe rephrase?

Comment (paulsavoie) [reply src/main/java/io/jenkins/plugins/signpath/SignPathStepBase.java:175]: I specified the error message like that. I don't understand your phrase @Taha-cmd

File: src/main/java/io/jenkins/plugins/signpath/SignPathStepBase.java (hunk 1 above)
Comment (volbobvol) [reply src/main/java/io/jenkins/plugins/signpath/SignPathStepBase.java]: Right, fixed

**Review by Taha-cmd — APPROVED** [review]
